finagle-core/src/main/scala/com/twitter/finagle/context/MarshalledContext.scala

```diff
@@ -130,9 +130,9 @@ final class MarshalledContext private[context] extends Context {
 
   // Exposed for testing
   private[context] def marshal(env: Map[Buf, Cell]): Iterable[(Buf, Buf)] = {
-    env.mapValues {
-      case Real(k, v) => k.marshal(v.x)
-      case Translucent(_, vBuf) => vBuf
+    env.transform {
```

Comment:
todo: What?

Reply:
this is a bummer.  should we keep it mapValues and suppress deprecations?  or add some kind of TODO to change it back after the new mapValues is rolled out?

Reply:
`transform` is AFAIK the fastest of the current options. I have no strong opinions on the matter, other than that unrelatedly the first case should maybe have been `case (_, Real(k, Some(v)) => k.marshal(v)`

Reply:
yeah, that makes sense–but presumably mapValues will be cheaper than transform when it's reimplemented, since it won't be wrapping the kv pairs in tuples, so I'd like a reminder to switch back to mapValues when it exists (hence the TODO).

I'd be down to switch it, except that I don't know if the Some unapply gets desugared to allocate an Option?  my hunch is that case class unapply is special and avoids it, but idk.  as an aside, I was looking at the API, and it looks like we can use .value instead of .get if we know it's a `Some`, which `Real` already guarantees for us.  should we use .value?

Reply:
case class unapply is indeed special and doesn't have to allocate anything. `value` was introduced in `2.12`, `get` is needed for 2.11 compat. `get` probably gets jitted to direct field access pretty quickly, so from a performance perspective it doesn't matter much. But if you're still pining for a reason to get rid of 2.11, this may be your ticket -- but again, I'm fine with whatever.

Reply:
if case class unapply is special, let's use it.  :shipit: